docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,373 @@
+****[//]: # (title: Compatibility guide for Kotlin 1.7)
+
+_[Keeping the Language Modern](kotlin-evolution.md)_ and _[Comfortable Updates](kotlin-evolution.md)_ are among the fundamental principles in
+Kotlin Language Design. The former says that constructs which obstruct language evolution should be removed, and the
+latter says that this removal should be well-communicated beforehand to make code migration as smooth as possible.
+
+While most of the language changes were already announced through other channels, like update changelogs or compiler
+warnings, this document summarizes them all, providing a complete reference for migration from Kotlin 1.6 to Kotlin 1.7.
+
+## Basic terms
+
+In this document we introduce several kinds of compatibility:
+
+- _source_: source-incompatible change stops code that used to compile fine (without errors or warnings) from compiling
+  anymore
+- _binary_: two binary artifacts are said to be binary-compatible if interchanging them doesn't lead to loading or
+  linkage errors
+- _behavioral_: a change is said to be behavioral-incompatible if the same program demonstrates different behavior
+  before and after applying the change
+
+Remember that those definitions are given only for pure Kotlin. Compatibility of Kotlin code from the other languages
+perspective
+(for example, from Java) is out of the scope of this document.
+
+## Language
+
+<!--
+### Title
+
+> **Issue**: [KT-NNNNN](https://youtrack.jetbrains.com/issue/KT-NNNNN)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**:
+>
+> **Deprecation cycle**:
+>
+> - 1.5.20: warning
+> - 1.7.0: report an error
+-->
+
+### Make safe call result always nullable
+
+> **Issue**: [KT-46860](https://youtrack.jetbrains.com/issue/KT-46860)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.7 will consider the type of safe call result always nullable, even when the receiver of the safe call is non-nullable.
+>
+> **Deprecation cycle**:
+>
+> - \<1.3: report a warning on unnecessary safe call on non-nullable receivers
+> - 1.6.20: warn additionally that the result of unnecessary safe call will change its type in the next version
+> - 1.7.0: change the type of safe call result to nullable,  
```

Comment:
```suggestion
> - 1.7.0: change the type of safe call result to nullable,
```

Reply:
As always, this trailing space is needed to have a soft break before the next line

Reply:
Do you use our documentation plugin? Its local preview shows that these trailing spaces don't make a line break, but in fact leave no space between the comma and the flag name
If it actually works correct on staging, I don't mind these extra spaces

Reply:
Hmm, it is supposed to work this way in markdown. What is the way then to insert a line break in webhelp md dialect?